### PR TITLE
rfc6555: init at 0.0.0

### PR DESCRIPTION
**Pull request by endocrimes**
###### Motivation for this change

rfc6555 is a required dependency of OfflineIMAP 7.3.0.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

### CODE REVIEW

**Review by spacekookie — APPROVED** [review]
LGTM!

**Comment by d-goldin** [comment]
Briefly tested it with python 2.7 and 3.7 on NixOS and seems fine in both cases. Maybe it would be nice to briefly comment in the derivation why tests are disabled.

**Comment by endocrimes** [comment]
@d-goldin Can do - they're currently disabled bc they require networking - is it preferable to keep them disabled or is there a way to allow networking during the `checkPhase`?

**Comment by d-goldin** [comment]
@endocrimes: Yeah, I tried it before and arrived at the same conclusion. Just thought it might make sense to leave a remark in the code so it can be patched in the future or just inform. 

To my knowledge wrt. network access - while there might be some ways to hack around the sandbox, we shouldn't go there, as it would potentially sacrifice purity reproducibility. But also better to not keep hitting external resources from the build-infra. 

So often such tests are either selectively disabled/removed or patched to not rely on external resources (ideally upstream of course).

**Comment by endocrimes** [comment]
@d-goldin :+1: added a comment

**Comment by vcunat** [comment]
> is there a way to allow networking during the `checkPhase`?

No, the nix language still doesn't have a good way for that AFAIK.  Well, you could cheat by creating a separate fixed-output derivation that produces something trivial on success (e.g. an empty file) and doing tests within that, but... there are also good arguments to try avoiding tests that require internet access.  I believe that localhost networking should work fine already (even in the sandbox).

**Comment by endocrimes** [comment]
@vcunat I’m well aware of the downsides of running tests that require network access 😅 - in this case I’m packaging a depending that I am not a maintainer of, so I have limited influence over that. In this case disabling checks is probably fine though.

**Comment by vcunat** [comment]
If these are just exceptions, it might be worth trying to disable just those that need internet.

**Comment by d-goldin** [comment]
I'm not sure if the disabled tests are holding this PR back, but in case it helps, here is a commit that more selectively disables tests by using a pre-existing decorator: https://github.com/d-goldin/nixpkgs/commit/f796e1e732fc3046af973ea726f7466cf881a727

**Review by vcunat — APPROVED** [review]
Seems fine, though I can't say I know python stuff well.

**Comment by vcunat** [comment]
I took your patch with my nitpicks as b199e30680e.  At a quick look, it might be worth submitting [upstream](https://github.com/sethmlarson/rfc6555), if you like.

**Comment by d-goldin** [comment]
@vcunat: Why `doCheck` alone was insufficient, is because it runs `${python.interpreter} setup.py test` per default, but for that a `test_suite` should be defined in `setup.py`. What is done in the upstream repo instead is that they use `tox`to kick-off the tests and run for different "environments" (https://github.com/sethmlarson/rfc6555/blob/master/tox.ini#L8). For our purposes here trying to use tox would have been more complicated than just adjusting the phase.

Thanks.

**Comment by vcunat** [comment]
Here I did `checkPhase` -> `postCheck`, so that we only extend the logic instead of overriding it.  Long-term experience says it's much better that way.

**Pull request merged**
